src/mpl/configure.ac

```diff
@@ -1086,7 +1086,22 @@ fi
 if test "X${pac_have_hip}" = "Xyes" ; then
     have_gpu="yes"
     AC_DEFINE([HAVE_HIP],[1],[Define if HIP is available])
-    AC_DEFINE([__HIP_PLATFORM_AMD__],[1],[AMD GPU HIP available])
+    AS_IF([test -n "${with_hip_include}"],
+          [PAC_APPEND_FLAG([-I${with_hip_include}],[CPPFLAGS])],
+          [AS_IF([test -n "${with_hip}"],
+                 [PAC_APPEND_FLAG([-I${with_hip}/include],[CPPFLAGS])])])
+
+    AS_IF([test -n "${with_hip_lib}"],
+          [PAC_APPEND_FLAG([-L${with_hip_lib}],[LDFLAGS])],
+          [AS_IF([test -n "${with_hip}"],
+                 [PAC_APPEND_FLAG([-L${with_hip}/lib],[LDFLAGS])
+                  AS_IF([test -d "${with_hip}/lib64"],
+                       [PAC_APPEND_FLAG([-L${with_hip}/lib64],[LDFLAGS])])
+                 ])
+          ])
+    PAC_APPEND_FLAG([-D__HIP_PLATFORM_AMD__], [CPPFLAGS])
+    PAC_LIBS_ADD(-lamdhip64)
+    GPU_SUPPORT="HIP"
```

Comment:
Why do we need `PAC_PUSH_FLAG` in the first place?

Reply:
The `hip_runtime_api.h` requires it to be defined for compilation. It is usually defined by hipcc compiler. Since we are not using it we need to manually define it. The PUSH/POP is simply for protecting the `CPPFLAGS`, so that we don't pollute it when hip is not available.

Reply:
The new code you added is equivalent to not using `PAC_PUSH_FLAG` in the first place, right?

Reply:
Yes. It is sets up the CPPFLAGS and LIBS the same way as the code between PAC_PUSH_FLAG and PAC_POP_FLAG.

Reply:
Then just remove `PAC_PUSH/POP_FLAG`, wouldn't that solve the issue?

Reply:
removing the PAC_PUSH/POP_FLAG also works fine to build mpich with hip libraries linked.

Reply:
@hzhou I guess another solution would be adding parameter to `PAC_CHECK_HEADER_LIB_OPTIONAL` and handle the extra CPPFLAGS internally. In this way, the `configure.ac` can be simple and clean. What do you think?

Reply:
@yfguo Didn't @sudheer9 confirm that simply remove `PAC_PUSH/POP_FLAG` works?

Reply:
@hzhou Yes. But removing the PUSH/POP will end up with `CPPFLAGS` containing `-D__HIIP_PLATFORM_AMD__` when HIP is not available.

Reply:
Then just remove `-D__HIP_PLATFORM_AMD__` when hip is not available. You can do this by using `sed`